src/Core/AdminConsole/OrganizationFeatures/Policies/Implementations/SavePolicyCommand.cs

```diff
@@ -0,0 +1,114 @@
+            {
+                throw new Exception($"Duplicate PolicyDefinition for {policyDefinition.Type} policy.");
+            }
+        }
+
+        _policyDefinitions = policyDefinitionsDict;
+    }
+
+    public async Task SaveAsync(Policy policy, IOrganizationService organizationService, Guid? savingUserId)
+    {
+        var org = await _applicationCacheService.GetOrganizationAbilityAsync(policy.OrganizationId);
+        if (org == null)
+        {
+            throw new BadRequestException("Organization not found");
+        }
+
+        if (!org.UsePolicies)
+        {
+            throw new BadRequestException("This organization cannot use policies.");
+        }
+
+        var policyDefinition = GetDefinition(policy.Type);
+        var allSavedPolicies = await _policyRepository.GetManyByOrganizationIdAsync(org.Id);
+        var currentPolicy = allSavedPolicies.SingleOrDefault(p => p.Id == policy.Id);
+
+        // If enabling this policy - check that all policy requirements are satisfied
+        if (currentPolicy is not { Enabled: true } && policy.Enabled)
+        {
+            var missingRequiredPolicyTypes = policyDefinition.RequiredPolicies
+                .Where(requiredPolicyType =>
+                    allSavedPolicies.SingleOrDefault(p => p.Type == requiredPolicyType) is not { Enabled: true })
+                .ToList();
+
+            if (missingRequiredPolicyTypes.Count != 0)
+            {
+                // TODO: would be better to reference the name instead of the enum
+                throw new BadRequestException("Policy requires PolicyType " + missingRequiredPolicyTypes.First() + " to be enabled first.");
+            }
+        }
+
+        // If disabling this policy - ensure it's not required by any other policy
+        if (currentPolicy is { Enabled: true } && !policy.Enabled)
+        {
+            var dependentPolicies = _policyDefinitions.Values
+                .Where(policyDef => policyDef.RequiredPolicies.Contains(policy.Type))
+                .Select(policyDef => policyDef.Type)
+                .Select(otherPolicyType => allSavedPolicies.SingleOrDefault(p => p.Type == otherPolicyType))
+                .Where(otherPolicy => otherPolicy is { Enabled: true })
+                .ToList();
+
+            if (dependentPolicies is { Count: > 0 })
+            {
+                throw new BadRequestException("This policy is required by " + dependentPolicies.First()!.Type + " policy. Try disabling that policy first.");
+            }
+        }
+
+        // Run other validation
+        var validationError = await policyDefinition.ValidateAsync(currentPolicy, policy);
+        if (!string.IsNullOrEmpty(validationError))
+        {
+            throw new BadRequestException(validationError);
+        }
+
+        // Run side effects
+        await policyDefinition.OnSaveSideEffectsAsync(currentPolicy, policy, organizationService);
+
+        policy.RevisionDate = DateTime.UtcNow;
```

Comment:
We don't currently use this, but we should. We have some fairly ugly hacks like [AssertRecent](https://github.com/bitwarden/server/blob/6f840758e57e1d24d3a6e81f13a4e6b8df9a0ad5/test/Common/Helpers/AssertHelper.cs#L228-L237) instead.

Given that this is a change to our current pattern I'm going to double check in `#code` before introducing it here, just to make sure it won't cause any issues as I'm not very familiar with TimeProvider.

Reply:
I was looking over the Passwordless code and we don't inject it anywhere so I believe its just built in. So it should be alright to start leveraging it.  `TimeProvider` does return `DateTimeOffset` instead of `DateTime` but that's easy enough to just do `_timeProvider.GetUtcNow().UtcDateTime` instead of `DateTime.UtcNow`.

Reply:
This is done!